cinder/scheduler/filter_scheduler.py

```diff
@@ -111,6 +111,33 @@ def schedule_create_volume(self, context, request_spec, filter_properties):
                                          filter_properties,
                                          allow_reschedule=True)
 
+    def find_backend_for_extend(self, context, request_spec,
+                                filter_properties):
+        volume_properties = request_spec['volume_properties']
+
+        if (volume_properties.get('consistencygroup_id')
+                or volume_properties.get('group_id')):
+            raise exception.NoValidBackend(
+                reason='The volume is in a group and cannot be migrated.')
+
+        # CapacityFilter needs to filter by the full size
+        filter_p = filter_properties.copy()
+        new_size = filter_p.pop('new_size')
+        request_spec['volume_properties']['size'] = new_size
```

Comment:
Need to make sure we are honoring the affinity/anti-affinity settings on the volume

https://github.com/sapcc/cinder/blob/stable/train-m3/cinder/volume/api.py#L225-L238

https://github.com/sapcc/cinder/blob/stable/train-m3/cinder/volume/api.py#L861-L862